src/js/index.js

```diff
@@ -39,3 +39,205 @@ function addTOCInteractivity() {
 $(document).ready(() => {
   addTOCInteractivity();
 });
+
+
+function setupVersionSwitcher() {
+  // Setup Version Switcher
+
+  // Only enable version switcher when window.versionSwitcher is filled by sphinx
+  if (!window.versionSwitcher) {
+    return;
+  }
+  let pageName = window.versionSwitcher.pageName;
+  let versionJsonUrl = window.versionSwitcher.versionJsonUrl;
+  let enableLocaleSupport = window.versionSwitcher.enableLocaleSupport;
+  let allLocales = window.versionSwitcher.allLocales;
+
+  // Remote version should like this.
+  // .name and .alias must be unique in each locale.
+  // It's not necessary to have same versions in all locales.
+  // When locale is enabled, there must be only one .default=true item in each locale to indicate which one should be redirect if target version doesn't exist in target locale.
+
+  /* 
+      let allVersions = {
+        "en": [
+          {
+            "name": "v1.2.0",
+            "url": "v1.2.0",
+            "alias": ["latest"],
+            "default": true
+          },
+          {
+            "name": "v1.1.0",
+            "url": "v1.1.0",
+            "alias": []
+          },
+        ],
+        "zh":[
+          {
+            "name": "v1.0.0",
+            "url": "v1.0.0",
+            "alias": []
+            "default": true
+          },
+        ],
+      };
+      */
+  function parseCurrentURL() {
+    // parseCurrentURL look up current pathname, generate all information about current version and locale.
+
+    let pathname = window.location.pathname;
+
+    // add "index.html" back when browser omit it.
+    if (pageName.endsWith("index.html") && pathname.endsWith("/")) {
+      pathname += "index.html";
+    }
+    if (pathname.slice(-pageName.length) !== pageName) {
+      // Sphinx generated pages should have exactly same suffix
+      throw 'page suffix do not match requirements';
```

Comment:
Do we want to `throw` here (and below)? Or do, we want to just fail gracefully somehow. If any of the parsing failed, maybe remove the version switcher element altogether? Or fill it with a go-to-home button instead. I'm not sure what the best path is.